src/apps/relay/userdb.c

```diff
@@ -68,6 +68,9 @@ static TURN_MUTEX_DECLARE(o_to_realm_mutex);
 static ur_string_map *o_to_realm = NULL;
 static secrets_list_t realms_list;
 
+int global_allocation_count = 0; // used for drain mode, to know when all allocations have gone away
```

Comment:
Why use a mutex protected into over an atomic int? An atomic into would be much less cumbersome to use and faster to access.

Reply:
Also, use size_t over into for "count" of things.

Reply:
I wasn't sure that there was existing an atomic increment in C that all your supported compilers would be happy with.  Do you know what might supported by all OSs?

Good idea for size_t change, will do.

Reply:
Shouldn't the C11 atomic functions work?

https://en.cppreference.com/w/c/atomic

I don't know of an official list of supported platforms for coturn. I'm not a maintainer.

That said, personally, if it's not in the CI system, i don't consider it a supported platform. Someone who cares about a platform that doesn't (somehow...?) have C11 atomics missed their chance to get a CI runner / action setup to ensure their needs are met.

They can always use older releases.

Reply:
I can try it out and see how the CI builds go.

Reply:
Well it turns out the windows CI builds are not happy with stdatomic.h use.  I get this error when trying to use it in my VS2022:
The contents of <stdatomic.h> are available only with C++23 or later.

I will revert this back to mutex use tomorrow, unless someone prefers conditional code for windows vs non-windows.

Reply:
My understanding of the visual studio compiler is that it is exclusively a c++ compiler with very minor affordances for c code.

You could use std::atomic, which my work uses in c++ on visual studio with no issue.

It has an interface much nicer than the interlocked functions.